avocado/utils/network/interfaces.py

```diff
@@ -388,52 +400,92 @@ def save(self, ipaddr, netmask):
         current_distro = distro_detect()
 
         filename = f"ifcfg-{self.name}"
+        prefix = self.netmask_to_cidr(netmask)
         if current_distro.name in ["rhel", "fedora"]:
-            path = "/etc/sysconfig/network-scripts"
+            if(current_distro.name == 'rhel' and current_distro.version == '9'):
+                filename = f"{self.name}.nmconnection"
+                path = "/etc/NetworkManager/system-connections"
+            else:
+                path = "/etc/sysconfig/network-scripts"
         elif current_distro.name == "SuSE":
             path = "/etc/sysconfig/network"
         else:
             msg = "Distro not supported by API. Could not save ipaddr."
             raise NWException(msg)
 
-        ifcfg_dict = {
-            "TYPE": self.if_type,
-            "BOOTPROTO": "static",
-            "NAME": self.name,
-            "DEVICE": self.name,
-            "ONBOOT": "yes",
-            "IPADDR": ipaddr,
-            "NETMASK": netmask,
-            "IPV6INIT": "yes",
-            "IPV6_AUTOCONF": "yes",
-            "IPV6_DEFROUTE": "yes",
-        }
-        if current_distro.name == "SuSE":
-            ifcfg_dict.pop("BOOTPROTO")
+        if(current_distro.name == 'rhel' and current_distro.version == '9'):
+            ifcfg_dict = ''
+            if os.path.exists(f"{path}/{filename}"):
+                pass
+            else:
+                cmd1 = f"nmcli connection add con-name {self.name} ifname {self.name} type ethernet ipv4.address {ipaddr}/{prefix}"
+                run_command(cmd1, self.host)
```

Comment:
why not just have command line itself part of run_command

Reply:
@PraveenPenguin  You mean instead of having cmd1 as parameter, i need to add the command inside in run_command?

Please let me know am i understanding it in correct way?

Reply:
yes as we can avoid to having extra variable